messagecatalog/spec.md

```diff
@@ -0,0 +1,1042 @@
+
+The message definition version URI is composed from the
+[authority](#authority-definition-authority) (as the base URI) and the
+[message definition version id](#id-definition-version-id).
+
+In the default case, this URI corresponds directly to the URI of the
+["getMessage definitionVersion"](#342-get-a-specific-schema-version) API
+operation, which is convenient for simple usage scenarios.
+
+In [replication scenarios](#4-replication-model-and-state-change-events), the
+URI might not correspond to a resolvable network address, as permitted for the
+authority URI in the previous section. A consumer SHOULD then be configured with
+a fixed message catalog endpoint for it to use, and obtain definitions
+identified by a URI using the
+["getMessage definitionVersionByURI"](#342-get-a-specific-schema-version) API
+operation on that endpoint, rather than trying to resolve the URI directly.
+
+The following table shows some examples for a catalog hosted at
+`https://example.com/messagecatalog`:
+
+| [authority](#authority-definition-authority) | [id](#id-definition-version-id)                               | Message definition URI                                                                         |
+| -------------------------------------------- | ------------------------------------------------------------- | ---------------------------------------------------------------------------------------------- |
+| (empty, therefore implied)                   | `342`                                                         | <https://example.com/messagecatalog/342>                                                       |
+| <https://example.com/messagecatalog>         | `342`                                                         | `https://example.com/messagecatelog/342`                                                       |
+| `http://messagecatalog.example.com/`         | `/mygroup/messagedefinitions/myMessage definition/versions/2` | <http://messagecatalog.example.com/mygroup/messagedefinitions/myMessage definition/versions/2> |
+| `http://messagecatalog.example.org/`         | `http://messagedefinitions.example.org/41751`                 | <http://messagecatalog.example.org/41751>                                                      |
+| `urn:example:messagecatalog`                 | `a1b2c3d4`                                                    | `urn:example:messagecatalog:a1b2c3d4`                                                          |
+
+### 2.2.4. Message definition attributes
+
+The message definition object holds a collection of message definition versions.
+An implementation MAY add further attributes.
+
+#### `id` (message definition)
+
+- Type: `String`
+- Description: Identifies the message definition.
+- Constraints:
+  - REQUIRED. IMMUTABLE.
+  - MUST be a non-empty string
+  - MUST conform with RFC3986/3.3 `segment-nz-nc` syntax
+  - MUST be unique within the scope of the message definition group
```

Comment:
How does this work in replication scenarios? Are only complete _message definition groups_ replicated? Otherwise there might be collisions during replication, if message message definitions from multiple catalogs or groups are aggregated into a single group.